src/framework/mpas_pool_routines.F

```diff
@@ -5992,5 +5992,30 @@ integer function pool_get_member_decomp_type(dimName) result(decompType)!{{{
 
    end function pool_get_member_decomp_type!}}}
 
+   subroutine pool_hash(hash, key)!{{{
+
+      use iso_c_binding, only : c_int, c_char
+      use mpas_c_interfacing, only : mpas_f_to_c_string
+
+      implicit none
+
+      interface
+        subroutine c_pool_hash(hash, key) bind(c)
+           use iso_c_binding, only : c_int, c_char
+           integer (c_int), intent(inout) :: hash
+           character (c_char), dimension(*), intent(in) :: key
+        end subroutine c_pool_hash
+      end interface
+
+      integer (c_int), intent(inout) :: hash
+      character(len=*), intent(in) :: key
+
+      character(len=StrKIND, kind=c_char) :: c_key
```

Comment:
For consistency, I'd propose we declare this as:
```
character(kind=c_char), dimension(StrKIND+1) :: c_key
```
Note that the `+1` is probably important in the length to account for a C null character that will be added by `mpas_f_to_c_string`.